### PR TITLE
Feature/contract init

### PR DESCRIPTION
**Pull request by AddressXception**
add contract initialization that supports minting to an address

fixes: #7 #9

### CODE REVIEW

File: test/box.initialize.test.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+            wallet
+        }
+    })
+
+    it('Should mint specified nubmer of tokens to address', async () => {
+        const { box, wallet } = testContext;
+        await box.initialize(wallet.address, 10);
+
+        expect((await box.totalSupply()).toNumber()).to.equal(10);
+        expect((await box.balanceOf(wallet.address)).toNumber()).to.equal(10);
```
Comment (CryptoSpaces) [inline test/box.initialize.test.ts:31]: Believe the `.toNumber()` is not necessary with this matcher

File: test/box.purchase.test.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    describe('Happy flow', () => {
+      it('Should allow purchase of box', async () => {
+        const { box } = testContext
+        const boxBuyers = new Array(10)
+
+        for (let i = 0; i < boxBuyers.length; i++) {
+          const price: BigNumber = await box.getPrice()
+          await box.purchase({ value: price })
+
+          expect((await box.totalSupply()).toNumber()).to.equal(i + 1)
```
Comment (CryptoSpaces) [inline test/box.purchase.test.ts:57]: Same here `toNumber` is not necessary. 
I know I put these here, but it's unnecessary :D

File: test/box.purchase.test.ts [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+      expect(await box.maxSupply()).to.equal(bc.MAX_CURVE_VALUE)
+    })
+  })
+
+  describe('Purchase a box', () => {
+    describe('Happy flow', () => {
+      it('Should allow purchase of box', async () => {
+        const { box } = testContext
+        const boxBuyers = new Array(10)
```
Comment (CryptoSpaces) [inline test/box.purchase.test.ts:51]: I had this set up differently first.
This array is not needed. Could simply be a number `10`

File: test/box.purchase.test.ts [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+        for (let i = 0; i < boxBuyers.length; i++) {
```
Comment (CryptoSpaces) [inline test/box.purchase.test.ts:53]: and then make this `boxBuyers`

File: contracts/token/RarePizzasBox.sol [hunk 5]
```diff
@@ -29,18 +29,30 @@ contract RarePizzasBox is
...
+    string public _boxMetadataUri =
         "https://ipfs.io/ipfs/some/path/to/box/metadata";
 
-    function initialize() public initializer {
+    CountersUpgradeable.Counter public _minted_pizza_count;
+    CountersUpgradeable.Counter public _purchased_pizza_count;
+
+    // END V1 Variables
+
+    function initialize(address mintTo, uint8 count) public initializer {
```
Comment (CryptoSpaces) [inline contracts/token/RarePizzasBox.sol:49]: Should we check this address?

**Review by CryptoSpaces — APPROVED** [review]

File: contracts/token/RarePizzasBox.sol (hunk 5 above)
Comment (AddressXception) [reply contracts/token/RarePizzasBox.sol:49]: what do you mean check it?  this function gets called on contract deployment so I am certainly assuming that the deployer knows what they are doing.  i was actually considering removing the mint functionality and just forcing it to be called separately

**Pull request merged**